From e928f75ca45df8fb170230c95860f1a0fb3ae4e9 Mon Sep 17 00:00:00 2001
From: Justin Lecher <jlec@gentoo.org>
Date: Tue, 05 Jun 2012 08:17:24 +0000
Subject: Fix underlinking detected by gold linker #677445

Gold's strong underlinking detection showed missing libs during linking.
Gnome bug #677445

Signed-off-by: Justin Lecher <jlec@gentoo.org>
---
diff --git a/src/Makefile.am b/src/Makefile.am
index fe1e9ba..9e2f40d 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -2,7 +2,7 @@ libexec_PROGRAMS = libsocialweb-core
 
 libsocialweb_core_CFLAGS = -I$(top_srcdir) $(DBUS_GLIB_CFLAGS) $(SOUP_CFLAGS) $(GCOV_CFLAGS)
 libsocialweb_core_SOURCES = main.c poll.h poll.c
-libsocialweb_core_LDADD = ../libsocialweb/libsocialweb.la $(DBUS_GLIB_LIBS) $(SOUP_LIBS) $(GCOV_LDFLAGS)
+libsocialweb_core_LDADD = ../libsocialweb/libsocialweb.la $(DBUS_GLIB_LIBS) $(SOUP_LIBS) $(GCOV_LDFLAGS) $(GLIB_LIBS)
 
 %.service: %.service.in ../config.log
 	$(AM_V_GEN)sed -e "s|\@dir\@|$(libexecdir)|" $< > $@
diff --git a/tests/Makefile.am b/tests/Makefile.am
index e40cafd..59aa612 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -2,7 +2,7 @@ noinst_PROGRAMS = test-online test-client-online test-download test-download-asy
 
 test_online_SOURCES = test-online.c
 test_online_CFLAGS = -I$(top_srcdir) $(GOBJECT_CFLAGS)
-test_online_LDADD = $(GOBJECT_LIBS) ../libsocialweb/libsocialweb.la
+test_online_LDADD = $(GOBJECT_LIBS) ../libsocialweb/libsocialweb.la $(GLIB_LIBS)
 
 test_client_online_SOURCES = test-client-online.c
 test_client_online_CFLAGS = -I$(top_srcdir) $(GOBJECT_CFLAGS)
@@ -10,12 +10,12 @@ test_client_online_LDADD = $(GOBJECT_LIBS) ../libsocialweb-client/libsocialweb-c
 
 test_download_SOURCES = test-download.c
 test_download_CFLAGS = -I$(top_srcdir) $(GOBJECT_CFLAGS) $(SOUP_CFLAGS) $(DBUS_GLIB_CFLAGS)
-test_download_LDADD = $(GOBJECT_LIBS) $(SOUP_LIBS) ../libsocialweb/libsocialweb.la
+test_download_LDADD = $(GOBJECT_LIBS) $(SOUP_LIBS) ../libsocialweb/libsocialweb.la $(GLIB_LIBS)
 
 test_download_async_SOURCES = test-download-async.c
 test_download_async_CFLAGS = -I$(top_srcdir) $(GOBJECT_CFLAGS) $(SOUP_CFLAGS) $(DBUS_GLIB_CFLAGS)
-test_download_async_LDADD = $(GOBJECT_LIBS) $(SOUP_LIBS) ../libsocialweb/libsocialweb.la
+test_download_async_LDADD = $(GOBJECT_LIBS) $(SOUP_LIBS) ../libsocialweb/libsocialweb.la $(GLIB_LIBS)
 
 test_upload_SOURCES = test-upload.c
 test_upload_CFLAGS = -I$(top_srcdir) $(GOBJECT_CFLAGS) $(SOUP_CFLAGS) $(DBUS_GLIB_CFLAGS)
-test_upload_LDADD = $(GOBJECT_LIBS) $(SOUP_LIBS) ../libsocialweb/libsocialweb.la ../libsocialweb-client/libsocialweb-client.la
+test_upload_LDADD = $(GOBJECT_LIBS) $(SOUP_LIBS) ../libsocialweb/libsocialweb.la ../libsocialweb-client/libsocialweb-client.la $(GLIB_LIBS)
--
cgit v0.9.0.2

